zio-http-gen/src/main/scala/zio/http/gen/openapi/EndpointGen.scala

```diff
@@ -925,8 +928,10 @@ final case class EndpointGen(config: Config) {
             enums = Nil,
           ),
         )
-
-      case JsonSchema.Enum(enums) =>
+      case JsonSchema.Object(_, _, _)                                                                  =>
+        // properties.isEmpty && additionalProperties.isRight
+        throw new IllegalArgumentException("Top-level maps are not supported")
```

Comment:
@987Nabil why not actually?
If we want to derive an endpoint of a type like: `Endpoint[Unit, Unit, Nothing, Map[String, String], None]`,
how should we go about to do so?

Reply:
This type would then need to be generated inline. But this code is for generating a file, not an inline type.